src/olympia/devhub/views.py

```diff
@@ -1490,32 +1490,33 @@ def submit_finish(request, addon_id, addon):
     # Bounce to the versions page if they don't have any versions.
     if not addon.versions.exists():
         return redirect(addon.get_dev_url('versions'))
-    sp = addon.current_version.supported_platforms
-    is_platform_specific = sp != [amo.PLATFORM_ALL]
+    uploaded_version = addon.versions.latest()
+    supported_platforms = uploaded_version.supported_platforms
+    is_platform_specific = supported_platforms != [amo.PLATFORM_ALL]
 
     try:
         author = addon.authors.all()[0]
     except IndexError:
         # This should never happen.
         author = None
 
-    if author:
-        submitted_addons = (author.addons
-                            .exclude(status=amo.STATUS_NULL).count())
-        if submitted_addons == 1:
-            # We can use locale-prefixed URLs because the submitter probably
-            # speaks the same language by the time he/she reads the email.
-            context = {
-                'app': unicode(request.APP.pretty),
-                'detail_url': absolutify(addon.get_url_path()),
-                'version_url': absolutify(addon.get_dev_url('versions')),
-                'edit_url': absolutify(addon.get_dev_url('edit')),
-                'full_review': addon.status == amo.STATUS_NOMINATED
-            }
-            tasks.send_welcome_email.delay(addon.id, [author.email], context)
+    if (author and not author.addons.exclude(status=amo.STATUS_NULL)
```

Comment:
yeah, if they have at least one Addon with a version.channel==RELEASE_CHANNEL_LISTED then don't send them an email.  Ideally check if version.status isn't STATUS_NULL too, which will send dupes like we do now but will avoid the problem of not sending any email if their first attempt at submitting a version is aborted before finish (which I guess is fairly common)